Makefile.features

```diff
@@ -22,20 +22,34 @@ include $(RIOTCPU)/$(CPU)/Makefile.features
 # Resolve FEATURES_ variables
 # Their value will only be complete after resolving dependencies
 
-# Features that are required by the application but not provided by the BSP and
-# features that are used but blacklisted (prepended with "!").
-# Having features missing may case the build to fail.
-FEATURES_MISSING = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_REQUIRED)))
-
 # Features that are only optional and not required at the same time.
 # The policy is to by default use by features if they are provided by the BSP.
 FEATURES_OPTIONAL_ONLY = $(sort $(filter-out $(FEATURES_REQUIRED),$(FEATURES_OPTIONAL)))
 FEATURES_OPTIONAL_USED = $(sort $(filter $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
 # Optional features that will not be used because they are not provided
 FEATURES_OPTIONAL_MISSING = $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
 
+# Features that are used without taking "one out of" dependencies into account
+FEATURES_USED_SO_FAR = $(sort $(FEATURES_REQUIRED) $(FEATURES_OPTIONAL_USED))
+
+# Additionally required features due to the "one out of" dependencies
+FEATURES_REQUIRED_ADDITIONS = $(foreach ooo,$(FEATURES_REQUIRED_ANY),$(call _features_any,$(subst |, ,$(ooo)),$(ooo)))
+
+# Returns:
+# - If one (or more) of the features in $1 is already used
+#   (listed in FEATURES_USED_SO_FAR): An empty string
+# - If none of the features in $1 is used:
+#    - If none of the features in $1 is provided (not counting blacklisted
+#      features): $2
+#    - Else: One features that is both provided and listed in $1
+_features_any = $(if $(filter $1,$(FEATURES_USED_SO_FAR)),,$(word 1,$(filter $1,$(filter-out $(FEATURES_BLACKLIST),$(FEATURES_PROVIDED))) $2))
```

Comment:
I think this would be better with intermediate steps so something like (untested!):

```
/* FEATURES_REQUIRED_ANY that already have been resolved, there is one in  FEATURES_USED_SO_FAR */
_features_any_resolved =  $(filter $1,$(FEATURES_USED_SO_FAR))

/* return first feature in list that is `PROVIDED` and not `BLACKLISTED, can be empty `/
_features_use_first_one = $(firstword $(filter $1,$(filter-out $(FEATURES_BLACKLIST),$(FEATURES_PROVIDED))))

/* return nothing is requirements is fulfilled, first provided requirement in list or $2 if requirement cant be fulfilled */
_features_any = $(if $(call_features_any_resolved, $1),,$(firstword $(call _features_use_first_one ,$1) $2))
```

I think we could also have:

`FEATURES_PROVIDED_USABLE = $(filter-out $(FEATURES_BLACKLIST),$(FEATURES_PROVIDED))` and use that instead of `FEATURES_PROVIDED`.

Reply:
I added two intermediate steps as suggested, but a little different. Please check!